src/uproot/source/fsspec.py

```diff
@@ -148,28 +123,49 @@ def chunks(
         self._num_requested_chunks += len(ranges)
         self._num_requested_bytes += sum(stop - start for start, stop in ranges)
 
+        try:
+            # not available in python 3.8
+            to_thread = asyncio.to_thread
+        except AttributeError:
+            import contextvars
+            import functools
+
+            async def to_thread(func, /, *args, **kwargs):
+                loop = asyncio.get_running_loop()
+                ctx = contextvars.copy_context()
+                func_call = functools.partial(ctx.run, func, *args, **kwargs)
+                return await loop.run_in_executor(None, func_call)
```

Comment:
To be honest I just copied the implementation of `to_threads` for python 3.8 where it's not implemented (https://github.com/python/cpython/blob/main/Lib/asyncio/threads.py#L12-L25), I think "my" implementation and the upstream one should do the same.

From the documentation (https://docs.python.org/3/library/asyncio-task.html#running-in-threads) I assumed calling `to_thread` would run the coroutine in a separate thread, so if you call it 100 times, it will spawn 100 different threads, but reading it more carefuly I don't this this is the case, it's just using the default `concurrent.futures.ThreadPoolExecutor` with however many threads this has which I think is: `min(32, os.cpu_count() + 4)`.

I think this is still a good way to paralelize the requests and it removes from uproot the responsability to decide the number of threads to spawn.